### DISCUSSION TITLE
Usage in RESTful API bearer only mode

### DISCUSSION THREAD

**Discussion by eugeniosegala** [panva/openid-client#621] [Q&A] [answered]
Hi!

I have noticed that the `Strategy` that it being offered as a part of `node-openid-client` [requires a session to work](https://github.com/panva/node-openid-client/blob/main/lib/passport_strategy.js#L82).

I think this is perfectly fine for stateful services however it will not work for steteless Node JS APIs where a session is not present.

I have seen other [less popular libraries](https://github.com/hgranlund/passport-keycloak-bearer/tree/master) solving this problem in combination with Passport.

It this possible at all with `node-openid-client`?

Maybe there are other helpers/approaches which I'm missing.

Thanks!

**Comment by panva** [comment, marked as answer]
This is a login strategy, not an API bearer strategy. https://www.npmjs.com/package/express-oauth2-jwt-bearer might be what you're looking for if your access tokens are in the JWT format.
